### ISSUE TITLE
feat: support loading icons using http

### ISSUE THREAD

**Issue by HyperLife1119** [ng-icons/ng-icons#71]
Add a new URL configuration to allow loading icons using HTTP.

Similar to dynamic loading in NG-ZORRO, we can configure a URL from which the icon will be dynamically loaded
https://ng.ant.design/components/icon/en#static-loading-and-dynamic-loading

```ts
provideDynamicIcons({
   url: '/assets/icons'
})
```

Not only NG-ZORRO, ionicon also supports dynamic loading of icons:
https://github.com/ionic-team/ionicons/blob/main/src/components/icon/request.ts

**Comment by ashley-hunter** [comment]
Hi, thanks for the suggestion, what was it you had in mind exactly, something like this?

```ts
provideIcons({
   add: () => {
        // in here you could return a promise or an observable to lazy load the add icon svg
   }
})
```

What would you expect to happen while the icon is loading, but the element is visible on the screen? Should it remain blank?

**Comment by HyperLife1119** [comment]
Sorry, my previous description was not clear enough, I just updated the feature description!

**Comment by ashley-hunter** [comment]
Thanks, that helps. I think in our case we won't be able to list the URL directly as our icons are simply strings so there is no fast way to differentiate between an icon and a URL. We could however do something like I mentioned above, allow icons to be defined as functions that return a promise or observable. This would give us some nice flexibility in how icons can be imported, for example:

```ts
provideIcons({
   // dynamic imports - lazy loading a local file
   add: () => import('./icon-file').then(m => m.Add),
   // using fetch
   minus: () => fetch('https://example.com/minus.svg').then(m => m.text())
   // using Angular HTTP client
   divide: (http) => http.get('https://example.com/divide.svg')
})
```

This way we can do a simple `typeof` check to know whether an icon is a string, i.e. a static icon, or a function which is a icon that needs to be loaded.

**Comment by HyperLife1119** [comment]
I think this requires providing a whole new `provideDynamicIcons` function:

```ts
provideDynamicIcons({
   icon: 'ionicon' | 'bootstrap' | '...',
   url: 'url'
})
```

With this feature, we no longer need to manually import each icon, we only need to specify which icon set and icon URL to use.

**Comment by ashley-hunter** [comment]
Ok, gotcha, so you are proposing being able to specify a "base url" for icons that would be fetched on the fly. Sounds reasonable, I'll look into it later tonight, thanks

**Comment by ashley-hunter** [comment]
Thanks, I have now implemented this functionality and it is available in v25.5.0:

https://github.com/ng-icons/ng-icons#dynamically-loading-icons

Any issues just let me know, thanks

**Comment by HyperLife1119** [comment]
Thanks, I'm already using this feature and feel good about it.

**Comment by HyperLife1119** [comment]
It would be better if there was a built-in icon cache, which I currently need to do:

```ts
provideNgIconLoader((() => {
  const map = new Map<string, string>();

  return async (name: string) => {
    let svg = map.get(name);
    if (svg) return svg;

    const resp = await fetch(`/assets/icons/${name}.svg`);
    svg = await resp.text();
    map.set(name, SVG);
    return svg;
   }
})())
```

Maybe we can add a feature function to selectively enable:

```ts
provideNgIconLoader(
  name => ...,
  withCaching()
)
```

What do you think?

**Comment by ashley-hunter** [comment]
Good idea! I can certainly add that.

**Comment by ashley-hunter** [comment]
Caching is now available in the latest release. Thanks for the suggestions:

<img width="861" alt="image" src="https://github.com/ng-icons/ng-icons/assets/20795331/7e8235a2-06c8-46b4-ad02-faa48b6173bb">

**Comment by HyperLife1119** [comment]
Thanks, this looks good!

**Comment by HyperLife1119** [comment]
```html
@for (item of array; track item) {
  <ng-icon name="checkmark-outline"></ng-icon>
}
```

In list loop rendering, multiple HTTP requests for the same icon will appear. I think we should also need to cache ongoing requests to ensure that the same request is used when multiple requests for the same icon are made at the same time.

**Comment by ashley-hunter** [comment]
Thanks for the suggestion, this should now be resolved in 26.1.0
